packages/eslint-config-basic/index.js

```diff
@@ -1,3 +1,41 @@
+const fs = require('node:fs')
+
+// only files with extension below will be checked
+const VALID_EXTENSION = [
+  'js', 'cjs', 'mjs', 'jsx',
+  'ts', 'cts', 'mts', 'tsx',
+  'json', 'json5',
+  'yaml', 'yml',
+  'html', 'vue',
+  'md',
+]
+
+function isInvalidFile(path) {
+  return !VALID_EXTENSION.some(ext => path.toLowerCase().endsWith(`.${ext}`))
+}
+
+function isFile(path) {
+  let stats
+  try {
+    stats = fs.statSync(path)
+  }
+  catch (e) {
+    return false
+  }
+  return stats.isFile()
+}
+
+// ignore for lint-staged
+const ignorePatterns = process.argv
+  .slice(2)
```

Comment:
I am not very enjoying this approach, I honestly prefer to statically list them. And maybe on the other hand, usages on lint-stage would be better to explicitly set the extension (we could update the readme)

Reply:
Hi. Is there any reasons that you prefer to statically list them? For example, `*.sh` files are widely used in enterprise app. Users have to manually add some configs to ignore them (by editing `.eslinttc` or adding `.eslintginore` file). It's not convenient.

As for working with `lint-staged`, user don't know which type of file will be linted by this eslint config. They have to check the source code. And then setup `lint-staged.config.json` to `{"*.js,*.vue,*.md":"eslint --fix"}`, it's not clean enough for user.

Ignoring those we cannot lint is reasonable in my humble opinion.

Reply:
Yes, I agree we should ignore un-lint-able files, and that's how we are doing now.

My concern is that your approach is unstable and inefficient. It's saying that the configuration will be different based on your input, or usages like `eslint .` will be missing even the common extensions. While you see that `eslint .` already automatically filters out supported files. To me I think it would be best for ESLint to support ignoring unknown extensions, like [`--ignore-unknown` in Prettier](https://prettier.io/docs/en/cli.html#--ignore-unknown).